src/components/expression/expression-renderer.js

```diff
@@ -459,58 +620,53 @@ genotet.ExpressionRenderer.prototype.drawGeneProfiles_ = function() {
   if (!this.data.options.showProfiles) {
     this.svgProfile_.selectAll('*').attr('display', 'none');
     return;
-  }
-  else {
+  } else {
     if (!this.data.options.showGeneLabels) {
-      this.PROFILE_MARGINS_.LEFT = this.DEFAULT_PROFILE_MARGIN;
-    }
-    else {
-      this.PROFILE_MARGINS_.LEFT = this.geneLabelWidth_ + this.LABEL_MARGIN_;
+      this.profileMargins_.left = this.DEFAULT_PROFILE_MARGIN;
+    } else {
+      this.profileMargins_.left = this.geneLabelWidth_ + this.LABEL_MARGIN;
     }
     this.svgProfile_.selectAll('*').attr('display', 'inline');
   }
 
   var heatmapData = this.data.matrix;
-  this.svgProfile_.selectAll('g').remove();
   this.svgProfile_.attr('width', this.canvasWidth_);
 
+  this.svgLegend_
```

Comment:
Setting positions of svg groups could be moved to layout(), so does the computation of group translation, e.g. this.profileMargins_.left here.